INSTALL

```diff
@@ -535,11 +535,14 @@ The specific libraries and their options are:
   Beginning with MFEM v3.3, SUNDIALS v2.7.0 is supported.
   Beginning with MFEM v3.3.2, SUNDIALS v3.0.0 is also supported.
   Beginning with MFEM v4.1, only SUNDIALS v5.0.0+ is supported.
+  When MFEM_USE_CUDA is enabled, only SUNDIALS v5.4.0+ is supported.
```

Comment:
It looks like SUNDIALS v5.0.0 is not sufficient for a non-CUDA build. The internal tests on `toss3` give the following error with v5.0.0:
```console
../../libmfem.so: undefined reference to `N_VGetLocalVector_MPIPlusX'
```
When was `N_VGetLocalVector_MPIPlusX` introduced in SUNDIALS? If it was added before v5.4.0, we can build without CUDA with that older version but it is probably simpler to just say that we require v5.4.0+ now.

In this case, the above text and the version requirement below should be updated appropriately.

It is also a good idea to mention the new version requirement in the `CHANGELOG` in the new paragraph about SUNDIALS.

Reply:
`N_VGetLocalVector_MPIPlusX` is in SUNDIALS `v5.0.0` so I am not sure why that error is occurring. Can you share details about how SUNDIALS and mfem were built in this case? I am able to do a non CUDA build on my box with v5.0.0.

Reply:
Okay, I think you are right -- the problem is on our side. In the internal tests we explicitly set the SUNDIALS libraries with which to link, overriding the defaults, and we need to update our scripts to include `-lsundials_nvecmpiplusx`.

Reply:
This reminds me: we'll need to update the mfem spack package accordingly here: https://github.com/spack/spack/blob/59720e0ef9b28a6bc1b7f2deee7e079ce378da02/var/spack/repos/builtin/packages/mfem/package.py#L676-L682. This will need to include `cvode` --> `cvodes` (from #1380), `nvecparhyp` --> `nvecmpiplusx`, and adding `nveccuda` when CUDA is enabled. On top of that, maintaining older mfem+sundials versions will complicate the logic even more. 🙁

Reply:
Since v5.0.0 does work without cuda I think all you need to do in the spack package is add something like

```
depends_on('sundials@5.4.0:+mpi+hypre+cuda', when='@4.2.0:+sundials+mpi+cuda')
```

and then 

```
@property
    def sundials_components(self):
        """Return the SUNDIALS components needed by MFEM."""
        sun_comps = 'arkode,cvode,nvecserial,kinsol'
        if '+mpi' in self.spec:
            sun_comps += ',nvecparhyp,nvecparallel,nvecmpiplusx'
        if '+cuda' in self.spec:
            sun_comps += ',nveccuda'
        return sun_comps
```